outwatch/src/main/scala/outwatch/util/Store.scala

```diff
@@ -14,58 +14,63 @@ import scala.util.control.NonFatal
 
 
 object Store {
+  sealed trait BaseAction
```

Comment:
I have to admit that I do not like it all that much either, but, in order to represent all 3 possible action types we would have to use something like...
```scala
  type Action = Option[Try[A]]
```

Since the previous action could also be an error. This would make the reducer look rather verbose:  
```scala
val reducer = {
  case (state, Some(Success(IncrementBy(number)))) => state += number
}
```

Reply:
I like having the `Action` as context to the `Throwable`. It would be rather useful when, for example, displaying a toast message to the user.